### PR TITLE
Add Container.is_port_open method

### PR DESCRIPTION
**Pull request by bkabrda**

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/499YT](https://da.gd/499YT)__
* Comment: __This is testing CI__

File: conu/utils/core.py [hunk 1]
```diff
@@ -100,14 +100,17 @@ def raw(self):
...
-            logger.debug('port OPENed: %s:%s' % (host, port))
-            return True
-        else:
-            logger.debug('port CLOSEDed: %s:%s' % (host, port))
-            return False
+        try:
+            sock.settimeout(timeout)
+            result = sock.connect_ex((host, port))
+            if result == 0:
+                logger.debug('port OPENed: %s:%s' % (host, port))
```
Comment (TomasTomecek) [inline conu/utils/core.py]: please improve the log messages, at least something like `port is opened: %s...`, `port is closed: %s...`

File: tests/integration/test_docker.py [hunk 2]
```diff
@@ -54,12 +54,14 @@ def test_networking_scenario():
     Listen via netcat in one container, send a secret message to the container via another one.
     """
     image = DockerImage(THE_HELPER_IMAGE)
-    r1 = DockerRunCommand(command=["nc", "-l", "0.0.0.0", "1234"])
+    r1 = DockerRunCommand(command=["nc", "-l", "-k", "0.0.0.0", "1234"])
```
Comment (TomasTomecek) [inline tests/integration/test_docker.py]: out of curiosity, does this test run okay for you locally?

File: conu/utils/core.py (hunk 1 above)
Comment (TomasTomecek) [reply conu/utils/core.py]: (`port CLOSEDed` shreds my eyes)

File: tests/integration/test_docker.py (hunk 2 above)
Comment (TomasTomecek) [reply tests/integration/test_docker.py]: b/c for me it fails like 50% of the time, so maybe `-k` will help

File: conu/utils/core.py (hunk 1 above)
Comment (bkabrda) [reply conu/utils/core.py]: Sure, I didn't actually touch the log messages in this PR, but I can fix them.

File: tests/integration/test_docker.py (hunk 2 above)
Comment (bkabrda) [reply tests/integration/test_docker.py]: It did succeed for me twice in a row and I didn't feel the need to do more runs, but I can try and see what's wrong there.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/f6YW](https://da.gd/f6YW)__
* Comment: __This is testing CI__

File: tests/integration/test_docker.py (hunk 2 above)
Comment (bkabrda) [reply tests/integration/test_docker.py]: I just tried running the test several times in a row, both isolated and as part of the whole test suite and it didn't fail even once. Could you possibly specify at which point the test is failing for you, and what seems to be wrong?

Comment (TomasTomecek) [reply tests/integration/test_docker.py]: It fails on the fact that the "secret" message won't reach the server netcat, so it fails on the assert. I would need to debug it with tcpdump locally and I'm lazy to do that, especially when it works for all of you and in CI.

File: tests/integration/test_docker.py [hunk 3]
```diff
@@ -75,6 +77,7 @@ def test_networking_scenario():
     cont2.popen_instance.communicate(input=secret_text + b"\n")
     # give container time to process
     time.sleep(1)
+    cont.stop()
     assert not cont2.is_running()
     assert not cont.is_running()
     assert secret_text == cont.logs().strip()
```
Comment (TomasTomecek) [inline tests/integration/test_docker.py]: @bkabrda this is the line where it usually fails for me

**Pull request merged**